File: leetcode/python/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        common = strs[0]
        
        i = 1

        while i < len(strs):
            if strs[i].startswith(common):
                i += 1
            else:
                common = common[:-1]
        
        print(common)
        return common

Solution.longestCommonPrefix(Solution, ["flower","flow","flight"])
        

File: leetcode/go/14-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    var common = ""

	for i := 0; i < len(strs[0]); i++ {
		for j := 0; j < len(strs); j++ {
			if i >= len(strs[j]) || strs[j][i] != strs[0][i] {
				return common
			}
		}
		common += string(strs[0][i])
	}

	return common
}